plugins/inputs/apcupsd/apcupsd_test.go

```diff
@@ -0,0 +1,129 @@
+package apcupsd
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestApcupsdGather(t *testing.T) {
```

Comment:
We need to split the tests out into individual test cases.  Each test should try to check only one thing, and similar tests can use a table driven test if possible.